src/main/java/com/synopsys/integration/detect/lifecycle/run/RunManager.java

```diff
@@ -188,6 +190,13 @@ private UniversalToolsResult runUniversalProjectTools(final PropertyConfiguratio
                 extractionEnvironmentProvider, codeLocationConverter, "DOCKER", DetectTool.DOCKER,
                 eventSystem);
             final DetectableToolResult detectableToolResult = detectableTool.execute(directoryManager.getSourceDirectory());
+            if (detectableToolResult.getFailedExtractableResult().isPresent()) {
+                //TODO: Remove hack when windows docker support added. This workaround allows docker to throw a user friendly exception when not-extractable due to operating system.
+                DetectableResult extractable = detectableToolResult.getFailedExtractableResult().get();
+                if (WrongOperatingSystemResult.class.isAssignableFrom(extractable.getClass())) {
+                    throw new DetectUserFriendlyException("Docker currently requires a non-Windows OS to run. Attempting to run Docker on Windows is not currently supported.", ExitCodeType.FAILURE_CONFIGURATION);
+                }
+            }
```

Comment:
Could we add a more permanent solution to support failures like this in the future?
Or is stopping on wrong os a one time thing?

Reply:
This also exceeds our cognitive complexity limit .
https://sonarcloud.io/project/issues?id=com.synopsys.integration%3Asynopsys-detect&open=AXGJcloCXEyONgMy3ji9&pullRequest=152&resolved=false&types=CODE_SMELL

Reply:
I'd argue we don't generally "stop", detect still tries to continue and do as much as it can. Only in this docker case have customers asked that detect stop, so I'd be hesitant to add a more permanent solution. Also this will eventually go away when Docker is supported on Windows.